roles/openshift_facts/library/openshift_facts.py

```diff
@@ -114,6 +114,12 @@ def migrate_hosted_facts(facts):
             if 'router' not in facts['hosted']:
                 facts['hosted']['router'] = {}
             facts['hosted']['router']['selector'] = facts['master'].pop('router_selector')
+        if 'registry_selector' in facts['master']:
+            if 'hosted' not in facts:
+                facts['hosted'] = {}
+            if 'registry' not in facts['hosted']:
+                facts['hosted']['registry'] = {}
+            facts['hosted']['registry']['selector'] = facts['master'].pop('registry_selector')
```

Comment:
To expand on this a bit more. If we do switch to not using openshift_facts here for the desired state, we would need to add introspection for the current state from the existing configured project node selectors and/or the node selectors defined on the DCs for the router/registry in particular.